flytekit/image_spec/image_spec.py

```diff
@@ -120,6 +122,48 @@ def exist(self) -> bool:
     def __hash__(self):
         return hash(asdict(self).__str__())
 
+    def run_commands(self, commands: Union[str, List[str]]):
+        """
+        Run a list of commands during the building process.
+        """
+        if self.commands is None:
```

Comment:
Lets not add any mutable apis to ImageSpec, ImageSpec should be immutable by default.

If you want to add new attributes, that should be like a builder.

ImageSpec.with_run_commands() -> ImageSpec

Reply:
done